src/simp.c

```diff
@@ -1187,6 +1187,9 @@ static tree_t simp_cond_assign(tree_t t)
    else {
       tree_t s = tree_new(T_IF);
       tree_set_loc(s, tree_loc(t));
+      
+      if (tree_has_ident(t))
+         tree_set_ident(s, tree_ident(t));
```

Comment:
In what circumstances is/isn't the label set here?

Reply:
You are right, it is set in all circumstances, both T_COND_ASSIGN and T_COND_VAR_ASSIGN will for sure have the label set as parser underneath creates the implicit label for both of them.